src/OpenTelemetry.Api/Trace/Source.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="Source.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Controls the source name and version that will be analyzed.
+    /// </summary>
+    public struct Source
```

Comment:
* `Source` is a very generic name. What about `TraceSource`?
* Probably don't need this to be a struct. It's only allocated on startup and then we keep the instance around in a dictionary right?

Reply:
ok. will change to class and rename to TraceSource :)
thanks for the feedback!

Reply:
Yes. The `ShouldListenTo` is executed only once, when an `ActivitySource` is newed up. So its not a hot path like `Sample`

Reply:
perfect. thanks for the explanation!